src/tools/pr_8799/src/virtcontainers.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::{anyhow, Result};
+use std::{fs, result::Result::Ok, path::Path, collections::HashMap};
+
+const CPU_POS_STAT: usize = 38;
+const PROC_STAT_SIZE: usize = 52;
+
+// create a map of tids-vcpus from /proc/pid subdirectories
+pub fn get_ch_vcpu_tids_by_path(proc_path: &str) -> Result<HashMap<u32,u32>> {
+
+    let mut tids = HashMap::new();
+
+    if !nix::unistd::Uid::effective().is_root() {
+       return Err(anyhow!("Will not perform pid checks as non root user"));
+    }
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("Path must be non-empty strings."));
+    }
+
+    // canonicalize the /proc/pid path path
+    let c_path = fs::canonicalize(proc_path)?;
+
+    // make the task path
+    let tid_path = fs::canonicalize(Path::new(&c_path).join("task"))?;
```

Comment:
I think you can drop this second call to `canonicalize` since at this point, `c_path` is already canonical so it's safe to append our own path element to it.